fs/zipfs.py

```diff
@@ -43,13 +44,27 @@
     R = typing.TypeVar("R", bound="ReadZipFS")
 
 
+def _bytes(s):
```

Comment:
Not keen on this dynamic handling of bytes/str. I know it seems like a convenience, but it tends to hide bugs. Better to check what format the Python zipfile library accepts and stick with that. If its different on Python versions, you may need an `if PY3` branch.

Reply:
Both py2 and py3 built-in `zipfile` lib only supports bytes for password but bytes is not intuitive for humans.
If this conversion is not accepted, I could and a type check and raise error if user not provides bytes password.

Reply:
> If this conversion is not accepted, I could and a type check and raise error if user not provides bytes password.

:+1:

The URL parameters would have to be an exception as they are always unicode/str.